src/stancjs/stancjs.ml

```diff
@@ -77,16 +80,41 @@ let stan2cpp model_name model_string is_flag_set flag_val =
                       canonicalizer_settings ) )
             , warnings
             , [] ) ;
+        if is_flag_set "debug-mir" then
+          r.return
+            ( Result.Ok
+                (Sexp.to_string_hum [%sexp (mir : Middle.Program.Typed.t)])
+            , warnings
+            , [] ) ;
         if is_flag_set "debug-generate-data" then
           r.return
             ( Result.Ok (Debug_data_generation.print_data_prog typed_ast)
             , warnings
             , [] ) ;
-        let mir = Ast_to_Mir.trans_prog model_name typed_ast in
-        let tx_mir = Transform_Mir.trans_prog mir in
         let opt_mir =
-          if is_flag_set "O" then Optimize.optimization_suite tx_mir else tx_mir
-        in
+          if is_flag_set "O0" then opt_lvl := Optimize.O0
+          else if is_flag_set "O1" then opt_lvl := Optimize.O1
+          else if is_flag_set "Oexperimental" || is_flag_set "O" then
+            opt_lvl := Optimize.Oexperimental ;
+          Optimize.optimization_suite
+            ~settings:(Optimize.level_optimizations !opt_lvl)
+            tx_mir in
+        if is_flag_set "debug-optimized-mir" then
+          r.return
+            ( Result.Ok
+                (Sexp.to_string_hum [%sexp (opt_mir : Middle.Program.Typed.t)])
+            , warnings
+            , [] ) ;
+        if is_flag_set "debug-transformed-mir" then
+          r.return
+            ( Result.Ok
+                (Sexp.to_string_hum [%sexp (tx_mir : Middle.Program.Typed.t)])
+            , warnings
+            , [] ) ;
+        if is_flag_set "dump-stan-math-signatures" then (
+          Stan_math_signatures.pretty_print_all_math_sigs Format.str_formatter
```

Comment:
This is just for debugging purposes as it would print signatures not return them. However this doesn't print anything. `List.length (Hashtbl.keys stan_math_signatures)` returns 0.

Reply:
I wonder if there is something about the module loading order in the JS runtime that means the initalizer code doesn't run. You could try refactoring the portion of stan_math_signatures that builds it into

```ocaml

let build_signatures () = (* current contents of the let () = ... block *)

let () = build_signatures ()


```

and then manually call Stan_math_signatures.build_signatures in stancjs before printing?

Reply:
Ah never mind, it's just the printing that doesn't seem to show up in the test. 

```ocaml
        if is_flag_set "dump-stan-math-signatures" then (
          let num_of_sigs = 
          Hashtbl.to_alist Stan_math_signatures.stan_math_signatures 
          |> List.length 
          |> string_of_int in
          r.return (Result.Ok num_of_sigs, warnings, []) ) ;
```
works just fine.